backend/test/deal-observer.test.js

```diff
@@ -74,3 +76,40 @@ describe('deal-observer-backend', () => {
     assert.deepStrictEqual(expected, actual)
   })
 })
+
+describe('deal-observer-backend binary', () => {
```

Comment:
Since this test is no longer testing the bin script, but only the loop, let's rename it accordingly:

```suggestion
describe('dealObserverLoop', () => {
```

I find it difficult to navigate in very large files, I prefer to have more smaller files instead. Since these tests are testing code from `backend/lib/deal-observer-loop.js`, I would like the test code to be in `backend/test/deal-observer-loop.test.js`. 

It's a convention we already use in other repositories. It makes it easy to find tests because we can navigate using file names instead of scanning a long test file containing tests for many different components to find the right `describe` block.